web/html/src/manager/virtualization/guests/list/guests-list.js

```diff
@@ -249,7 +250,7 @@ class GuestsList extends React.Component<Props, State> {
                   />)
                 }
                 {modalsData
-                  .filter(action => action.type !== 'delete' || this.props.saltEntitled)
+                  .filter(action => !this.props.foreignEntitled && (action.type !== 'delete' || this.props.saltEntitled))
```

Comment:
I wonder if it wouldn't be better to centralize this logic, having something like isActionVisible(action) used in both places